webhooks/virtualmachine/mutation/virtualmachine_mutator.go

```diff
@@ -310,24 +311,74 @@ func SetDefaultPowerState(
 	return false
 }
 
-// ResolveImageName mutates the vm.spec.imageName if it's not set to a vmi name
-// and there is a single namespace or cluster scope image with that status name.
+const (
+	vmiKind  = "VirtualMachineImage"
+	cvmiKind = "Cluster" + vmiKind
+)
+
+// ResolveImageName ensures vm.spec.image is set to a non-empty value if
+// vm.spec.imageName is also non-empty.
 func ResolveImageName(
 	ctx *context.WebhookRequestContext,
 	c client.Client,
 	vm *vmopv1.VirtualMachine) (bool, error) {
-	// Return early if the VM image name is empty or already set to a vmi name.
-	imageName := vm.Spec.ImageName
-	if imageName == "" || strings.HasPrefix(imageName, "vmi-") {
+
+	// Return early if the image reference is used.
+	if vm.Spec.Image != nil {
+		return false, nil
+	}
```

Comment:
What if `vm.Spec.Image` and `vm.Spec.imageName` are both set, but are pointing to different images? I don't see the validation webhook checking for this case. Should we check here in the mutation webhook and return an error in that case?

Reply:
I discussed that with @bryanv , and decided that `spec.image` is the authority. If it is set, `spec.imageName` is not used at all. Let me update the API docs to reflect that.